src/vmm_core.ml

```diff
@@ -54,42 +54,77 @@ module Name = struct
      name label -- in all these labels, only letters, digits, hyphens, and dot
      are allowed.
   *)
-  type path = string list
-  type t = path * string option
-
-  (* from OCaml 4.13 bytes.ml *)
-  let for_all p s =
-    let n = String.length s in
-    let rec loop i =
-      if i = n then true
-      else if p (String.unsafe_get s i) then loop (succ i)
-      else false in
-    loop 0
-
-  let [@inline always] valid_label s =
-    String.length s < 64 &&
-    String.length s > 0 &&
-    String.get s 0 <> '-' && (* leading may not be '-' *)
-    for_all (function
-        | 'a'..'z' | 'A'..'Z' | '0'..'9' | '-' | '.' -> true
-        | _ -> false)
-      s (* only LDH (letters, digits, hyphen, period)! *)
-
-  let path_equal (x, _) (y, _) =
-    let rec equal x y = match x, y with
-      | [], [] -> true
-      | x::xs, y::ys -> String.equal x y && equal xs ys
-      | _ -> false
-    in
-    equal x y
+  module Label = struct
+    type t = string
+
+    let [@inline always] valid s =
+      String.length s < 64 &&
+      String.length s > 0 &&
+      String.get s 0 <> '-' && (* leading may not be '-' *)
+      String.for_all (function
+          | 'a'..'z' | 'A'..'Z' | '0'..'9' | '-' | '.' -> true
+          | _ -> false)
+        s (* only LDH (letters, digits, hyphen, period)! *)
```

Comment:
with todays view, this restriction is a bit arbitrary. we may want our unikernels to be named 🛩️ or 😄 or 🍌... but we can wait a bit and figure out later (I'm not sure what other systems we interact with want as names..)